### PR TITLE
[JavaScript|Python] Adding try/catch to avoid exceptions crashing consumers

### PR DESCRIPTION
**Pull request by Grey0202**
-Add try/catch to JS .NET Python model file to avoid the crash.

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs [hunk 1]
```diff
@@ -19,17 +20,25 @@ protected ChoiceModel(IParser parser, IExtractor extractor)
...
         {
-            var extractResults = Extractor.Extract(query);
-            var parseResults = extractResults.Select(r => Parser.Parse(r));
+            try
+            {
+                var extractResults = Extractor.Extract(query);
+                var parseResults = extractResults.Select(r => Parser.Parse(r));
 
-            return parseResults.Select(pr => new ModelResult()
+                // o can be null, add this part to try.
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs]: What "o"?

File: .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs [hunk 2]
```diff
@@ -19,17 +20,25 @@ protected ChoiceModel(IParser parser, IExtractor extractor)
...
+            }
+            catch (Exception)
             {
-                Start = pr.Start.Value,
-                End = pr.Start.Value + pr.Length.Value - 1,
-                Resolution = GetResolution(pr),
-                Text = pr.Text,
-                TypeName = ModelTypeName,
-            }).ToList();
+                return null;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs]: Code should never return null. If no extracted entity, we should return an empty list.

File: JavaScript/packages/recognizers-choice/src/choice/models.ts [hunk 3]
```diff
@@ -24,6 +26,10 @@ export abstract class ChoiceModel implements IModel {
                 text: o.text,
                 typeName: this.modelTypeName
             }));
+        }
+        catch(err) {
+            return null;
```
Comment (tellarin) [inline JavaScript/packages/recognizers-choice/src/choice/models.ts]: Same.

File: JavaScript/packages/recognizers-choice/src/choice/models.ts [hunk 4]
```diff
@@ -12,10 +12,12 @@ export abstract class ChoiceModel implements IModel {
...
 
     parse(source: string): ModelResult[] {
-        let extractResults = this.extractor.extract(source);
-        let parseResults = extractResults.map(r => this.parser.parse(r));
+        try {
+            let extractResults = this.extractor.extract(source);
+            let parseResults = extractResults.map(r => this.parser.parse(r));
 
-        return parseResults
+            // o can be null, add this part to try.
```
Comment (tellarin) [inline JavaScript/packages/recognizers-choice/src/choice/models.ts]: How can it be null?

File: JavaScript/packages/recognizers-number-with-unit/src/numberWithUnit/models.ts [hunk 5]
```diff
@@ -21,48 +21,56 @@ export abstract class AbstractNumberWithUnitModel implements IModel {
...
-            let extractResults = extractor.extract(query);
-            let parseResults: Array<ParseResult> = [];
-            for (let i = 0; i < extractResults.length; i++) {
-                let r = parser.parse(extractResults[i]);
-                if (r.value !== null) {
-                    if (r.value instanceof Array) {
-                        for (let j = 0; j < r.value.length; j++) {
-                            parseResults.push(r.value[j]);
+
+        try{
```
Comment (tellarin) [inline JavaScript/packages/recognizers-number-with-unit/src/numberWithUnit/models.ts]: Shouldn't this type of try/catch be in the extract/parser level to make sure individual entity processings don't blow?

File: Python/libraries/recognizers-choice/recognizers_choice/choice/models.py [hunk 6]
```diff
@@ -16,18 +16,24 @@ def __init__(self, parser: Parser, extractor: Extractor):
...
-            model_result.text = o.text
-            model_result.type_name = self.model_type_name
-
-            result.append(model_result)
+
+        try:
+            extract_results = self.extractor.extract(source)
+            parse_results = [self.parser.parse(e) for e in extract_results]
+
+            # o can be null, add this part to try.
```
Comment (tellarin) [inline Python/libraries/recognizers-choice/recognizers_choice/choice/models.py]: "o" should never be null. When can this happen?

File: Python/samples/simple_console/sample.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
 from recognizers_suite import Culture, ModelResult
 
 # Use English for the Recognizers culture
-DEFAULT_CULTURE = Culture.English
+DEFAULT_CULTURE = Culture.Chinese
```
Comment (tellarin) [inline Python/samples/simple_console/sample.py]: Be careful not to commit such changes. Please revert this.

File: JavaScript/packages/recognizers-number-with-unit/src/numberWithUnit/models.ts (hunk 5 above)
Comment (Grey0202) [reply JavaScript/packages/recognizers-number-with-unit/src/numberWithUnit/models.ts]: > Shouldn't this type of try/catch be in the extract/parser level to make sure individual entity processings don't blow?

This part of .NET does the same thing.

File: .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs [hunk 8]
```diff
@@ -20,27 +20,30 @@ protected ChoiceModel(IParser parser, IExtractor extractor)
...
             {
-                return null;
+                // Nothing to do. Exceptions in parse should not break users of recognizers.
+                // No result.
             }
+
+            return parseResults.Select(pr => new ModelResult()
+            {
+                Start = pr.Start.Value,
+                End = pr.Start.Value + pr.Length.Value - 1,
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.Choice/Models/ChoiceModel.cs]: Won't this return an invalid object and the consuming code will need to filter these out?

**Pull request merged**